### DISCUSSION TITLE
Quarkus at JavaZone 2023

### DISCUSSION THREAD

**Discussion by insectengine** [quarkusio/quarkus#35360] [Events]
**Event Description:** The biggest community-driven Java conference.

**Date:** September 5-7, 2023
**Location:** Oslo, Norway
**Event Type:** In Person

https://2023.javazone.no/

### Quarkus Sessions:

**Session:** [Quarkus 3: The Road to Loom for Cheaper, Faster, and Easier Concurrent Applications](https://2023.javazone.no/program/808094c6-0893-4d9d-8fe2-533c6442c9b8)
**Speaker(s):** Daniel Oh
**Language:** English
**Day/Time:** Wednesday, September 06, 11:40 AM – 12:40 PM
**Abstract:** Developers have had extraordinary experiences and taken stunning capabilities while they were developing Java microservices from local to the cloud using Quarkus for the past 4 years. But there is more to come with their journey since we have just arrived in Quarkus 3 which provides good, better, and even excellent features and capabilities in terms of developer experience, performance, scalability, and cloud integration. Especially, Quarkus 3 simplifies asynchronous concurrent applications using virtual threads (Project Loom) for high scalability.

In this session, we take you through how Quarkus integrates Loom for developers to make concurrent applications easier, have cheaper memory, and have high performance using virtual threads. You can also learn about what’s new in Quarkus 3 such as JakartaEE 10, MicroProfile 6, Hibernate ORM 6, and more.

**Session:** [Quarkus for Spring Developers](https://2023.javazone.no/program/72a562c9-9cd9-43db-80e7-735ce5472de4)
**Speaker(s):** Eric Deandrea
**Language:** English
**Day/Time:** Thursday, September 07, 11:40 AM – 12:40 PM
**Abstract:** In this session, I will show concepts and conventions familiar to Spring developers and how those same concepts and conventions can be implemented in Quarkus, all while highlighting similarities and differences between them. Additionally, I will show similarities and differences in how testing is done, highlighting Quarkus Dev Services.
This session will be mostly code while minimizing the number of slides. I will introduce an existing Spring application with a full test suite and build a Quarkus equivalent version of it, live.

**Session:** [Cracking the scalability wall](https://2023.javazone.no/program/6bcfb467-baaf-4920-b11d-6603e0efd4dc)
**Speaker(s):** Sanne Grinovero
**Language:** English
**Day/Time:**  Thursday, 1:00 – 1:45 PM
**Abstract:** In this talk, we’ll discuss a severe JVM performance issue, the methodology leading to its discovery, how it affects the whole Java ecosystem, and what you could do to avoid it in your code.

This scalability bottleneck has existed in OpenJDK for decades; the most surprising aspect is how such an impactful problem has been undetected for so long, eluding existing diagnostic tooling, while many common idioms will trigger the issue. Chances are high that your code is affected as well.

After explaining the issue, we’ll show some real-world code from popular OSS libraries and how we worked around the problem in several popular OSS Java libraries with simple, small changes, leading to significant efficiency improvements.

### Quarkus Workshops

**Workshop:**  [Quarkus Deep Dive](https://2023.javazone.no/program/ec146183-6412-4751-a484-bd09e3bfaade)
**Speaker(s):** Kevin Dubois
**Language:** English
**Length:** 240 minutes
**Date/Time:** Tuesday, September 05, 9:00 AM – 1:00 PM
**Abstract:** Java is a great programming language, however 'traditional' Java isn't so great to work with when it comes to modern, Cloud Native development. Quarkus is a (fairly) new Java stack that addresses issues such as the typical slow startup time and rather large memory usage that hinder the adoption of Java in container and/or Serverless workloads. Quarkus is not just useful for optimizing resource usage though. There is also a big focus on improving the developer experience.

In this session we'll take a look at how Quarkus is very easy to work with and allows developers to work with containers and external dependencies such as databases, Kafka clusters, Kubernetes etc without being experts in any of these technologies. Not to mention that there is no need to manually recompile/redeploy to see any of these changes.

After this session, the audience should come away with inspiration to build modern Cloud Native applications with Java and Quarkus, and have fun doing so!

Prerequisites
We'll go through setup/signup instructions at the beginning of the lab. No specific requirements other than a decent working laptop :)

**Comment by iamgul** [comment]
Will it be available online? We from the different part of world want to experience the same so please make it available online as well.

**Reply by insectengine** [reply]
That is a question for the JavaZone staff... but I believe it's an in person event.

**Reply by edeandrea** [reply]
It's an in-person event organized by a core team of JUG leaders and other volunteers: https://2023.javazone.no/about
